src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/links/RoundTripIslSpec.groovy

```diff
@@ -0,0 +1,287 @@
+        }
+    }
+
+    @Tidy
+    def "A round trip latency ISL goes down when the switch supports round trip but rule for this purpose is not installed"() {
+        given: "A round trip latency ISL"
+        Isl roundTripIsl
+        def srcSwToDeactivate = topology.activeSwitches.find { sw ->
+            if (sw.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)) {
+                roundTripIsl = topology.getRelatedIsls(sw).find {
+                    it.dstSwitch.features.contains(SwitchFeature.NOVIFLOW_COPY_FIELD)
+                }
+                roundTripIsl
+            }
+        } ?: assumeTrue("Wasn't able to find a switch with suitable link", false)
+        def dstSw = roundTripIsl.dstSwitch
+
+        when: "Delete ROUND_TRIP_LATENCY_RULE_COOKIE on the dst switch"
+        northbound.deleteSwitchRules(dstSw.dpId, DeleteRulesAction.REMOVE_ROUND_TRIP_LATENCY)
+        def isRoundTripRuleDeleted = true
+        Wrappers.wait(RULES_DELETION_TIME) {
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.size() == 1
+        }
+
+        and: "Deactivate the src switch"
+        lockKeeper.knockoutSwitch(srcSwToDeactivate)
+        def isSrcSwDeactivated = true
+        // it takes more time to DEACTIVATE a switch via the 'knockoutSwitch' method on the stage env
+        Wrappers.wait(discoveryTimeout + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.DEACTIVATED
+        }
+
+        then: "The round trip latency ISL is FAILED"
+        sleep((discoveryTimeout + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.FAILED
+
+        when: "Install ROUND_TRIP_LATENCY_RULE_COOKIE on the dst switch"
+        northbound.installSwitchRules(dstSw.dpId, InstallRulesAction.INSTALL_ROUND_TRIP_LATENCY)
+        isRoundTripRuleDeleted = false
+        Wrappers.wait(RULES_INSTALLATION_TIME) {
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.empty
+        }
+
+        then: "The round trip latency ISL is still FAILED"
+        //ISL will be in the FAILED state until gets any discovery packet
+        sleep((discoveryInterval + WAIT_OFFSET / 2) * 1000 as long)
+        northbound.getLink(roundTripIsl).state == IslChangeType.FAILED
+
+        cleanup:
+        isSrcSwDeactivated && lockKeeper.reviveSwitch(srcSwToDeactivate)
+        isRoundTripRuleDeleted && northbound.installSwitchRules(dstSw.dpId, InstallRulesAction.INSTALL_ROUND_TRIP_LATENCY)
+        Wrappers.wait(discoveryInterval + WAIT_OFFSET) {
+            assert northbound.getSwitch(srcSwToDeactivate.dpId).state == SwitchChangeType.ACTIVATED
+            assert northbound.getAllLinks().findAll {
+                it.state == IslChangeType.DISCOVERED
+            }.size() == topology.islsForActiveSwitches.size() * 2
+            assert northbound.validateSwitch(dstSw.dpId).rules.missing.empty
+        }
+    }
```

Comment:
I don't understand the purpose of this test. The situation with 'missing RTL rule' is already like maximum strange. Ok, got 1 strange test for it. But this is a second test which does not test anything new over the ` "A round trip latency ISL goes down when the src switch is DEACTIVATED and round trip latency rule is removed \
on the dst switch"`. So I'm kinda confused by what is being tested here again

Reply:
you are right, there is no much sense here(
test is deleted